server/service/src/invoice_line/inbound_shipment_line/update/generate.rs

```diff
@@ -94,13 +95,14 @@ fn generate_line(
 
     if let Some(total_before_tax) = total_before_tax {
         update_line.total_before_tax = total_before_tax;
-    } else {
-        update_line.total_before_tax =
-            update_line.cost_price_per_pack * update_line.number_of_packs as f64;
-    }
-
-    if let Some(tax) = tax {
-        update_line.tax = tax.percentage;
+    } else if let Some(number_of_packs) = number_of_packs {
+        update_line.total_before_tax = update_line.cost_price_per_pack * number_of_packs as f64;
+    } else if let Some(cost_price_per_pack) = cost_price_per_pack {
+        update_line.total_before_tax = cost_price_per_pack * update_line.number_of_packs as f64;
+    } else if let (Some(number_of_packs), Some(cost_price_per_pack)) =
```

Comment:
this block will have to be first - if either `number_of_packs` or `cost_price_per_pack` aren't None then you won't reach the final if block - or is there some magic that I'm missing here?

Reply:
[0d8193d](https://github.com/openmsupply/open-msupply/pull/380/commits/0d8193d5e69eeaa1f10113ac6d3e607be3e60e7e)